yb-voyager/src/adaptiveparallelism/adaptive_parallelism.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package adaptiveparallelism
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	CPU_USAGE_USER                 = "cpu_usage_user"
+	CPU_USAGE_SYSTEM               = "cpu_usage_system"
+	MAX_CPU_THRESHOLD              = 70
+	ADAPTIVE_PARALLELISM_FREQUENCY = 10 * time.Second
+)
+
+type TargetYugabyteDBWithConnectionPool interface {
+	IsAdaptiveParallelismSupported() bool
+	GetClusterMetrics() (map[string]map[string]string, error) // node_uuid:metric_name:metric_value
+	GetNumConnectionsInPool() int
+	GetNumMaxConnectionsInPool() int
+	UpdateNumConnectionsInPool(int) error // (delta)
+}
+
+func AdaptParallelism(yb TargetYugabyteDBWithConnectionPool) error {
```

Comment:
I think this makes more sense in `tgtdb`?  similar to `conn_pool` and this function can be a `tdb` functionality that way we can call it using this `tdb.AdaptParallelism()` we will not implement it for other dbs.

Reply:
As we discussed, it is fine to not use `tdb` here and do type assertions, but see if its fine to move this to `tgtdb` package

Reply:
Works